### PR TITLE
Fix Table Update, duplicate column assignments, empty update $set/$unset 

### LINKED ISSUE #1728: HTTP 500 on fields overlapping in $set & $unset
```
CommandFailedEvent {
  command: {
    updateOne: {
      filter: { text: '92b038cd-e70e-4e47-88da-6dbc749c6a77', int: 0 },
      update: { '$set': { tinyint: 4 }, '$unset': { tinyint: '' } },
      options: { upsert: undefined }
    }
  },
  keyspace: 'default_keyspace',
  source: 'test_table',
  commandName: 'updateOne',
  url: 'https://c60822b2-b054-4874-978e-5c1c06ce8d54-us-west-2.apps.astra-dev.datastax.com/api/json/v1/default_keyspace/test_table',
  duration: 3071.164334,
  error: DataAPIHttpError: HTTP error (500): {"errors":[{"message":"An unexpected server error occurred while processing the request. \n\nError Class: SyntaxError\nError Message: Multiple incompatible setting of column tinyint","scope":"","errorCode":"UNEXPECTED_SERVER_ERROR","id":"9b6b8207-8bd5-40a7-b71a-9b53f3efc4c9","family":"SERVER","title":"Unexpected server error"}]}
      ... {
    status: 500,
    body: '{"errors":[{"message":"An unexpected server error occurred while processing the request. \\n\\nError Class: SyntaxError\\nError Message: Multiple incompatible setting of column tinyint","scope":"","errorCode":"UNEXPECTED_SERVER_ERROR","id":"9b6b8207-8bd5-40a7-b71a-9b53f3efc4c9","family":"SERVER","title":"Unexpected server error"}]}',
    raw: {
      headers: {
        server: 'openresty/1.25.3.1',
        date: 'Fri, 15 Nov 2024 06:20:10 GMT',
        'content-type': 'application/json;charset=UTF-8',
        'content-length': '330'
      },
      body: '{"errors":[{"message":"An unexpected server error occurred while processing the request. \\n\\nError Class: SyntaxError\\nError Message: Multiple incompatible setting of column tinyint","scope":"","errorCode":"UNEXPECTED_SERVER_ERROR","id":"9b6b8207-8bd5-40a7-b71a-9b53f3efc4c9","family":"SERVER","title":"Unexpected server error"}]}',
      status: 500,
      url: 'https://c60822b2-b054-4874-978e-5c1c06ce8d54-us-west-2.apps.astra-dev.datastax.com/api/json/v1/default_keyspace/test_table',
      httpVersion: 2,
      statusText: ''
    }
  }
}
```

### PR DESCRIPTION
**Pull request by Yuqi-Du**
This PR fixes two issues: #1728, #1727.
One PR for two issues is because they touched same file, and we try to keep it minimal and trackable for Dec Hot-Fix.

**Issue #1728 fix**
Duplicate column assignments in the update operation should fail.
E.G.
```json
{
    "updateOne": {
        "filter": {
            "id": "A2",
            "name": "Giraffe",
            "age": 5,
            "height": 450
        },
        "update": {
            "$set": {
                "description": "123"
            },
            "$unset": {
                 "description": "123"
            }
        }
    }
}
```

Previous:
`HTTP error (500): {"errors":[{"message":"An unexpected server error occurred while processing the request. \n\nError Class: SyntaxError\nError Message: Multiple incompatible setting of column tinyint","scope":"","errorCode":"UNEXPECTED_SERVER_ERROR"}`

Now:

```json
{
    "errors": [
        {
            "message": "The command included multiple update operations that attempted to change the same column. For example, attempting to both $set and $unset a column.\n\nMultiple assignments attempted to change the columns: description\n\nResend the command using a single update operation for each column.",
            "scope": "UPDATE",
            "errorCode": "UNSUPPORTED_OVERLAPPING_UPDATE_OPERATIONS",
            "id": "2638baff-a0a7-488d-84a5-57aa93f07ca9",
            "family": "REQUEST",
            "title": "Columns cannot be changed by multiple update operations"
        }
    ]
}
```

**Issue #1727 fix**
If an update does not any non-empty assignments operation, will error out.

e.g
So, these examples will fail with API exception 

- {"$set":{}}
- {"$unset":{}}
- {"$set":{}, "$unset":{}"}

This one won't, since there is a valid one
- {"$set":{"name":"yq"}, "$unset":{}"}

```json
{
    "errors": [
        {
            "message": "The command did not include any valid update operations to change the columns in the table. A valid update operation needs to have non-empty assignments.\n\nSupported update operations are $set, $unset.\n\nResend the command using at least one update operation.",
            "scope": "UPDATE",
            "errorCode": "MISSING_UPDATE_OPERATIONS",
            "id": "c4a2b180-6bf5-49d8-8842-3306b0df20b2",
            "family": "REQUEST",
            "title": "Update operation requires at least one operation"
        }
    ]
}
```

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #1728 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by amorton — CHANGES_REQUESTED** [review]
change to the error message, and it would be good to merge the tickets and fix for this and PR #1760 into one ticket and PR as they are changing the same class and it makes the changes for the hot fix release smaller

File: src/main/resources/errors.yaml [hunk 1]
```diff
@@ -397,6 +397,16 @@ request-errors:
       
       Resend the command using only supported update operations.
 
+  - scope: UPDATE
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/resources/errors.yaml:400]: solved thanks

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java [hunk 2]
```diff
@@ -97,6 +97,29 @@ public WithWarnings<UpdateValuesCQLClause> resolve(
...
+    if (!duplicates.isEmpty()) {
+      throw UpdateException.Code.DUPLICATE_UPDATE_OPERATION_ASSIGNMENTS.get(
+          errVars(
+              commandContext.schemaObject(),
+              map -> {
+                map.put(
+                    "duplicateAssignmentColumns",
+                    errFmtJoin(
+                        duplicates.stream()
+                            .map(CqlIdentifierUtil::cqlIdentifierToMessageString)
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java:118]: solved, thanks

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CommandQueryExecutor.java [hunk 3]
```diff
@@ -125,7 +125,6 @@ public Uni<AsyncResultSet> executeReadAllPages(
 
   public Uni<AsyncResultSet> executeWrite(SimpleStatement statement) {
     Objects.requireNonNull(statement, "statement must not be null");
-
```
[the remark this answers is not in the document]
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CommandQueryExecutor.java:128]: solved, thanks

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java [hunk 4]
```diff
@@ -97,6 +103,41 @@ public WithWarnings<UpdateValuesCQLClause> resolve(
...
+      throw UpdateException.Code.MISSING_UPDATE_OPERATIONS.get(
+          errVars(
+              commandContext.schemaObject(),
+              map -> {
+                map.put("supportedUpdateOperations", errFmtJoin(supportedOperatorsStringList));
+              }));
+    }
+
+    // Duplicate column assignments is invalid
+    // e.g. {"update": {"$set": {"description": "123"},"$unset": {"description": "456"}}}
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java:118]: Should we also handle subpaths? For example, `$set: { 'nested.path': 'test' }, $unset: { 'nested': 1 }`?

Another potential concern I had here was `$setOnInsert` and `$set` on the same field potentially being ok, but MongoDB also disallows that, so I think disallowing $setOnInsert and $set on the same field is reasonable.

Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java:118]: Thanks Val.
Currently only $set and $unset are supported in tables feature.

Comment (amorton) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateResolver.java:118]: and we do not have support for nested in tables, that may be how we handle maps and/or  UDT's

**Review by amorton — APPROVED** [review]
approved, made a small tweak to reduce duplicate code

**Comment by amorton** [comment]
ok to merge to main and to the hotifx for this one, because the duplicate assignments would produce unspecified results

**Pull request merged**